prow/github/hmac.go

```diff
@@ -46,3 +79,40 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// extractHmacs returns all *valid* HMAC tokens for given repository/organization. It iterates over full tree as necessary (repo->org->global)
+func extractHmacs(repo string, tokenGenerator func() []byte) ([][]byte, error) {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]hmacsForRepo{}
+
+	if err := yaml.Unmarshal(t, &repoToTokenMap); err != nil {
+		//NOTE : To keep backward compatibility, we are going to assume that in case of error, whole while is a single line hmac token.
+		//TODO : Once this code has been released and file has been moved to new format, we should delete this code and return error.
+		logrus.WithError(err).Error("Couldn't unmarshal the hmac secret as hierarchical file. Parsing as single token format.")
```

Comment:
This should be either reduced to info level, or provide immediately actionable information.

Reply:
I guess 'warning' should be an appropriate level. Also we don't expect this to be status quo for long because this would be removed as part of T+1 deployment.

Reply:
There are many prow deployments, most of which are not controlled by us. When changing things, we must provide for a substantial deprecation period with clear direction as to what must change.

(I would also argue we should deprecate things less than we do, since the benefit is usually minimal. In this case, some benefit does exist.)

Reply:
Given the context, a warning would still need actionable advice on resolving.

Reply:
fair enough.